### ISSUE TITLE
Error when extract to csv

### ISSUE THREAD

**Issue by mafsenra** [linville/pydbisam#4]
Hello, sorry to be bothering you. I need to get information from a table, which I send as an attachment, but I'm getting an error. I don't know enough about Python to resolve the error.
If you can help me, I would be very grateful.

Thanks.

[SLMESS.zip](https://github.com/linville/pydbisam/files/12714415/SLMESS.zip)

C:\Python3105>python
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()

C:\Python3105>pydbisam --dump-structure slmess.dat
Table: slmess.dat
  MD5 Checksum: 0x8ABE8E592364CB403D71D2E3BC64D001
  Last Updated: None
  Total Fields: 22
      Row Size: 168
    Total Rows: 4
  Deleted Rows: 0

  Column Info Offset: 512
         Data Offset: 17408
  Calc Row Area Size: 672
         File Length: 18080

Index, Name, Typ, Size, Row Offset
1, MESSMES, FieldType.INTEGER, 4, 26
2, MESSNCSS, FieldType.STRING, 13, 31
3, MESSCODCRSS, FieldType.STRING, 4, 45
4, MESSSITCRSS, FieldType.STRING, 4, 50
5, MESSTAXATOTAL, FieldType.FLOAT, 8, 55
6, MESSVALREM, FieldType.FLOAT, 8, 64
7, MESSVALCONT, FieldType.FLOAT, 8, 73
8, MESSNR, FieldType.INTEGER, 4, 82
9, MESSIC, FieldType.BOOLEAN, 1, 87
10, MESSVALFUNC, FieldType.FLOAT, 8, 90
11, MESSMESREF, FieldType.INTEGER, 4, 99
12, MESSANOREF, FieldType.INTEGER, 4, 104
13, MESSFICHAUTO, FieldType.BOOLEAN, 1, 109
14, MESSATUALIZADO, FieldType.BOOLEAN, 1, 112
15, UserID, FieldType.STRING, 16, 115
16, DataHora, FieldType.TIMESTAMP, 8, 132
17, MESSVALPAT, FieldType.FLOAT, 8, 141
18, MESS_COVID19, FieldType.BOOLEAN, 1, 150
19, MESS_LAYOFF, FieldType.BOOLEAN, 1, 153
20, MESS_FILESS_ID, FieldType.INTEGER, 4, 156
21, ID, FieldType.AUTOINCREMET, 8, 161
22, MESS_WS_FILESS, FieldType.BOOLEAN, 1, 166
------------------------------------------------------------------------------
C:\Python3105>pydbisam --dump-csv slmess.dat
MESSMES, MESSNCSS, MESSCODCRSS, MESSSITCRSS, MESSTAXATOTAL, MESSVALREM, MESSVALCONT, MESSNR, MESSIC, MESSVALFUNC, MESSMESREF, MESSANOREF, MESSFICHAUTO, MESSATUALIZADO, UserID, DataHora, MESSVALPAT, MESS_COVID19, MESS_LAYOFF, MESS_FILESS_ID, ID, MESS_WS_FILESS
Traceback (most recent call last):
  File "C:\Python3105\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python3105\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Python3105\Scripts\pydbisam.exe\__main__.py", line 7, in <module>
  File "C:\Python3105\lib\site-packages\cli.py", line 28, in main
    for row in db.rows():
  File "C:\Python3105\lib\site-packages\pydbisam\__init__.py", line 104, in rows
    row_values = self.row(index)
  File "C:\Python3105\lib\site-packages\pydbisam\extract.py", line 137, in row
    return [field.decode_from_row(row_data) for field in self._columns]
  File "C:\Python3105\lib\site-packages\pydbisam\extract.py", line 137, in <listcomp>
    return [field.decode_from_row(row_data) for field in self._columns]
  File "C:\Python3105\lib\site-packages\pydbisam\field.py", line 150, in decode_from_row
    return struct.unpack("<i", field_data)[0]
struct.error: unpack requires a buffer of 4 bytes

**Comment by linville** [comment]
It's an error with how the `Id` auto-increment field is handled. Looks auto-increment fields may not have a leading marker.

If you quickly want to hack in the fix yourself, just modify https://github.com/linville/pydbisam/blob/4a4ebbbd98b993132da6eedd68f3e9dc08eca148/pydbisam/field.py#L149-L150

To look like this:

```Python
        elif self._type is FieldType.AUTOINCREMET:
            field_data = row_data[self.row_offset : self.row_offset + 4]
            return struct.unpack("<I", field_data)[0]
```

I was then able to dump the data. I'll push up a fix in a bit after I've done a bit more investigation (the DB you provided is the first one I've seen that uses an auto-increment field).

**Comment by mafsenra** [comment]
Thank you so much for your help. 
Just one more question: 
Is there a way for the dump to generate a csv file?

**Comment by linville** [comment]
The dump can't directly create the file but you can just use shell redirection to create the file:

`pydbisam --dump-csv slmess.dat > slmess.csv`

**Comment by mafsenra** [comment]
Good job. 
Once again, 
Thank you !!!

**Comment by linville** [comment]
Fixed in v1.2.0.
